driver-core/src/main/com/mongodb/spi/dns/DnsException.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.spi.dns;
+
+/**
+ * An exception indicating a DNS error;
+ *
+ * @since 4.6
+ * @see DnsClient#getAttributeValues(String, String)
+ */
+public class DnsException extends Exception {
```

Comment:
Thanks, I see. In this case we must also catch and wrap all unchecked exceptions from `DnsClient.getResourceRecordData` because neither there can be a guarantee that an implementation wraps all exceptions in `DnsException`, nor do we seem to require doing so via the documentation.

Reply:
However, as soon as we start catching all unchecked exceptions, which we must do to avoid leaking them out of the driver, making `DnsException` a checked exception brings nothing to the table.

Reply:
It's not an unchecked exception, and `Exception` is caught and wrapped